processor/metricstransformprocessor/datapoint_aggregation.go

```
@@ -0,0 +1,248 @@
+				timestampToPoints[p.Timestamp.Seconds] = append(points, p)
+			} else {
+				timestampToPoints[p.Timestamp.Seconds] = []*metricspb.Point{p}
+			}
+		}
+	}
+	return timestampToPoints, startTimestamp
+}
+
+// aggregatePoints aggregates points in the groups provided in timestampToPoints by the specific caluculation indicated by aggrType and dataType
+// Returns a group of aggregated points
+func (mtp *metricsTransformProcessor) aggregatePoints(timestampToPoints map[int64][]*metricspb.Point, aggrType AggregationType, dataType metricspb.MetricDescriptor_Type) []*metricspb.Point {
+	newPoints := make([]*metricspb.Point, 0, len(timestampToPoints))
+	for _, points := range timestampToPoints {
+		// use the timestamp from the first element because these all have the same timestamp as they are grouped by timestamp
+		timestamp := points[0].Timestamp
+		switch dataType {
+		case metricspb.MetricDescriptor_CUMULATIVE_INT64, metricspb.MetricDescriptor_GAUGE_INT64:
+			intPoint := mtp.computeInt64(points, aggrType)
+			newPoints = append(newPoints, &metricspb.Point{
+				Timestamp: timestamp,
+				Value:     intPoint,
+			})
+		case metricspb.MetricDescriptor_CUMULATIVE_DOUBLE, metricspb.MetricDescriptor_GAUGE_DOUBLE:
+			doublePoint := mtp.computeDouble(points, aggrType)
+			newPoints = append(newPoints, &metricspb.Point{
+				Timestamp: timestamp,
+				Value:     doublePoint,
+			})
+		case metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION, metricspb.MetricDescriptor_GAUGE_DISTRIBUTION:
+			if aggrType != Sum {
+				mtp.logger.Warn("Distribution data can only be aggregated by taking the sum")
+				break
+			}
+			distPoints := mtp.computeDistribution(points)
+			for _, p := range distPoints {
+				newPoints = append(newPoints, &metricspb.Point{
+					Timestamp: timestamp,
+					Value:     p,
+				})
+			}
+
+		}
+	}
+	return newPoints
+}
+
+// computeInt64 attempts to merge int64 points into one point based on the provided aggrType
```

You say attempt in a few places in the comments where its not necessary. If it was really an attempt then this function should return an error.